packages/x-date-pickers/src/DesktopNextDateTimePicker/DesktopNextDateTimePicker.tsx

```diff
@@ -60,7 +60,11 @@ const DesktopNextDateTimePicker = React.forwardRef(function DesktopNextDateTimeP
         label: defaultizedProps.label,
         ampm: defaultizedProps.ampm,
       }),
-    },
+      tabs: {
+        underlineLocation: 'top',
+        ...defaultizedProps.componentsProps?.tabs,
+      },
+    } as DesktopNextDateTimePickerProps<TDate>['componentsProps'],
```

Comment:
I've added casting as a least intrusive way of having type inference for props without re-engineering props typing to work with the hook below.

Reply:
Do we agree that the typing works fine without those casts ?
What are you trying to achieve with those casts ?


My main problem with keeping them without a clear gain, is that is reduces the type safety.
For example if I do a typo on a property I'm passing, without the cast it complains:

![image](https://user-images.githubusercontent.com/3309670/210975467-e10c2636-63e1-45d5-a702-ab6fbbd9a313.png)

With the cast it does not complain:

![image](https://user-images.githubusercontent.com/3309670/210975517-0ad3e34d-e94c-4a2f-b04c-a3cba3c1b1fb.png)


We had a real scenario a few days ago with the range toolbar where I had a typo in a variable name that was swallowed by the cast.

If we need to pass a prop that the slot does not know, I would at least put the cast at the slot level, on all the slots

![image](https://user-images.githubusercontent.com/3309670/210975872-cb4331bf-ca39-4003-841e-27a5b2e7b4d3.png)

Or even apply a single `@ts-ignore` to really avoid impacting the rest of the typing.

Reply:
Hm... Fair point. We can't have both benefits with as casting...
I'll try going with the new [satisfies operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html). Seems that it should do the trick of covering both requirements. 👌

Reply:
If it does that's a far better solution that anything we both proposed :star_struck:

Reply:
Need to clarify if it's safe to use this operator, but seems to be the ideal solution for such cases. 👍 
<img width="1298" alt="Screenshot 2023-01-06 at 17 21 49" src="https://user-images.githubusercontent.com/4941090/211042275-bda0bd3b-a3a5-42b1-a219-8b802b74a0f8.png">

Reply:
These `satisfies` do not end up in production code in any way.
They seem safe to use. 👌

Reply:
Will leave this place as is for the time being, because of a [known bug in next](https://github.com/vercel/next.js/issues/43799).
It works without problems in IDE as well as build, but dev and test do fail.